src/file_pipe_log/pipe_builder.rs

```diff
@@ -214,7 +214,10 @@ impl<F: FileSystem> DualPipesBuilder<F> {
         &mut self,
         machine_factory: &FA,
     ) -> Result<(M, M)> {
-        let threads = self.cfg.recovery_threads;
+        let threads = std::cmp::min(
```

Comment:
What if it's 0?

Reply:
It's checked in `Config::sanitize`.

Reply:
Then nothing needs to be recovered, our code handles zero gracefully, but I noticed rayon thread pool will treat 0 as default value though.